plugins/org.eclipse.n4js.transpiler.es/src/org/eclipse/n4js/transpiler/es/transform/ExpressionTransformation.xtend

```diff
@@ -219,11 +238,106 @@ class ExpressionTransformation extends Transformation {
 							}
 							if (value !== null) {
 								replace(propAccessExpr, _StringLiteral(value));
+								return true;
 							}
 						}
 					}
 				}
 			}
 		}
+		return false;
+	}
+
+	/**
+	 * Converts
+	 * <pre>a ?? b</pre>
+	 * to
+	 * <pre>($temp = a) != null ? $temp : b</pre>
+	 */
+	def private void transformCoalesceExpression(CoalesceExpression coalExpr) {
+		// convert
+		val tempVarSTE = addOrGetTemporaryVariable(CHAINING_COALESCING_TEMP_VAR_NAME, coalExpr);
+		val replacement = _ConditionalExpr(
+			_EqualityExpr(
+				_Parenthesis(
+					_AssignmentExpr(
+						_IdentRef(tempVarSTE),
+						coalExpr.expression
+					)
+				),
+				EqualityOperator.NEQ,
+				_NULL
+			),
+			_IdentRef(tempVarSTE),
+			coalExpr.defaultExpression
+		);
+		if (coalExpr.eContainer instanceof Expression) {
+			replace(coalExpr, _Parenthesis(replacement));
+		} else {
+			replace(coalExpr, replacement);
+		}
+	}
+
+	/**
+	 * Converts
+	 * <pre>a?.b.c</pre>
+	 * to
+	 * <pre>($temp = a) == null ? void 0 : $temp.b.c</pre>
```

Comment:
Is the `==null` check rewritten in subsequent step?

Reply:
If the missing check for undefined is what you mean:
the cool kidz use `x == null` for `x === void 0 || x === null` these days (as I was told), they're equivalent (I know ... it's Javascript 😄 ).

Reply:
Yes, that I was thinking about. Thank you for enlightening me, too :)